.github/workflows/tests.yml

```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - 'master'
+      - 'dev'
```

Comment:
I'm introducing a `dev` branch, because while this MR is an encapsulated unit of work, it will be much better to merge everything into master once at the end.